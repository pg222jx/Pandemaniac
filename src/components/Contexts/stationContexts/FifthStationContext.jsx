import React,{ useState, createContext } from 'react';

export const FifthStationContext = createContext()

export const FifthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FifthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FifthStationContext.Provider>
    );
}

