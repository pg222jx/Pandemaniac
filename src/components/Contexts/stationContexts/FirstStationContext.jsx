import React,{ useState, createContext } from 'react';

export const FirstStationContext = createContext()

export const FirstStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FirstStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FirstStationContext.Provider>
    );
}

