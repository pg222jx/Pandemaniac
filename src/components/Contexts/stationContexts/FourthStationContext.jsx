import React,{ useState, createContext } from 'react';

export const FourthStationContext = createContext()

export const FourthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <FourthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </FourthStationContext.Provider>
    );
}

