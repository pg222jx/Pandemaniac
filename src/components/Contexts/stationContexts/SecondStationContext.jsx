import React,{ useState, createContext } from 'react';

export const SecondStationContext = createContext()

export const SecondStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <SecondStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </SecondStationContext.Provider>
    );
}

