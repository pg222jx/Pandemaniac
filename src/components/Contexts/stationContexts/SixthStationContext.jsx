import React,{ useState, createContext } from 'react';

export const SixthStationContext = createContext()

export const SixthStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <SixthStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </SixthStationContext.Provider>
    );
}

