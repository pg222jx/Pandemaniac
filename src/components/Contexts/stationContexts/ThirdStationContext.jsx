import React,{ useState, createContext } from 'react';

export const ThirdStationContext = createContext()

export const ThirdStationProvider = props => {

    const [style, setStyle] = useState({bottom: undefined, left: undefined, visibility: 'hidden'})

    return (
        <ThirdStationContext.Provider value={[style, setStyle]}>
            {props.children}
        </ThirdStationContext.Provider>
    );
}

