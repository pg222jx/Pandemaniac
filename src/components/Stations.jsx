/**
 * 6 science centers.
 *
 * @author Pernilla GÃ¶th
 * @version 1.0.0
 */


import React, { useContext } from 'react'
import station from '../images/station.png'
import { FirstStationContext } from './Contexts/stationContexts/FirstStationContext'
import { SecondStationContext } from './Contexts/stationContexts/SecondStationContext'
import { ThirdStationContext } from './Contexts/stationContexts/ThirdStationContext'
import { FourthStationContext } from './Contexts/stationContexts/FourthStationContext'
import { FifthStationContext } from './Contexts/stationContexts/FifthStationContext'
import { SixthStationContext } from './Contexts/stationContexts/SixthStationContext'

const Player = () => {
    const [firstPosition, setFirstPosition] = useContext(FirstStationContext)// eslint-disable-line no-unused-vars
    const [secondPosition, setSecondPosition] = useContext(SecondStationContext)// eslint-disable-line no-unused-vars
    const [thirdPosition, setthirdPosition] = useContext(ThirdStationContext)// eslint-disable-line no-unused-vars
    const [fourthPosition, setFourthPosition] = useContext(FourthStationContext)// eslint-disable-line no-unused-vars
    const [fifthPosition, setFifthPosition] = useContext(FifthStationContext)// eslint-disable-line no-unused-vars
    const [sixthPosition, setsixthPosition] = useContext(SixthStationContext)// eslint-disable-line no-unused-vars

    return (
        <React.Fragment>
            <img src={station} alt="Science station for North America" id="firstStation" className="station" style={ firstPosition }/>
            <img src={station} alt="Science station for South America" id="secondStation" className="station" style={ secondPosition }/>
            <img src={station} alt="Science station for Africa" id="thirdStation" className="station" style={ thirdPosition }/>
            <img src={station} alt="Science station for Europe" id="fourthStation" className="station" style={ fourthPosition }/>
            <img src={station} alt="Science station for Asia" id="firthStation" className="station" style={ fifthPosition }/>
            <img src={station} alt="Science station for Australia" id="sixthStation" className="station" style={ sixthPosition }/>
        </React.Fragment>
    )
}
 
export default Player
