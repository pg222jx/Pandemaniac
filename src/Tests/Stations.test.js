import React from 'react'
import ReactDOM from 'react-dom'
import { render, screen } from '@testing-library/react';
import { shallow, mount } from 'enzyme'
import  { FirstStationContext, FirstStationProvider } from '../components/Contexts/stationContexts/FirstStationContext'
import  { SecondStationContext, SecondStationProvider } from '../components/Contexts/stationContexts/SecondStationContext'
import  { ThirdStationContext, ThirdStationProvider } from '../components/Contexts/stationContexts/ThirdStationContext'
import  { FourthStationContext, FourthStationProvider } from '../components/Contexts/stationContexts/FourthStationContext'
import  { FifthStationContext, FifthStationProvider } from '../components/Contexts/stationContexts/FifthStationContext'
import  { SixthStationContext, SixthStationProvider } from '../components/Contexts/stationContexts/SixthStationContext'
import Stations from '../components/Stations'

function renderStations(firstPosition, setFirstPosition, secondPosition, setSecondPosition, thirdPosition, setThirdPosition,
  fourthPosition, setFourthPosition, fifthPosition, setFifthPosition, sixthPosition, setSixthPosition) {
    return render(
      <FirstStationContext.Provider value={[firstPosition, setFirstPosition]}>
        <SecondStationContext.Provider value={[secondPosition, setSecondPosition]}>
            <ThirdStationContext.Provider value={[thirdPosition, setThirdPosition]}>
                <FourthStationContext.Provider value={[fourthPosition, setFourthPosition]}>
                    <FifthStationContext.Provider value={[fifthPosition, setFifthPosition]}>
                        <SixthStationContext.Provider value={[sixthPosition, setSixthPosition]}>
                            <Stations />
                        </SixthStationContext.Provider>
                    </FifthStationContext.Provider>
                </FourthStationContext.Provider>
            </ThirdStationContext.Provider>
        </SecondStationContext.Provider>
      </FirstStationContext.Provider>
    )
  }

  const wrapperShallow = shallow(
    <FirstStationProvider>
    <SecondStationProvider>
        <ThirdStationProvider>
            <FourthStationProvider>
                <FifthStationProvider>
                    <SixthStationProvider>
                        <Stations />
                    </SixthStationProvider>
                </FifthStationProvider>
            </FourthStationProvider>
        </ThirdStationProvider>
    </SecondStationProvider>
  </FirstStationProvider>
)

const wrapperMount = mount(
  <FirstStationProvider>
  <SecondStationProvider>
      <ThirdStationProvider>
          <FourthStationProvider>
              <FifthStationProvider>
                  <SixthStationProvider>
                      <Stations />
                  </SixthStationProvider>
              </FifthStationProvider>
          </FourthStationProvider>
      </ThirdStationProvider>
  </SecondStationProvider>
</FirstStationProvider>
)

describe('Testing the Stations component', () => {
  it('should match the snapshot', () => {
      expect(wrapperShallow.html()).toMatchSnapshot()
    })

  it('should render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <FirstStationProvider>
          <SecondStationProvider>
              <ThirdStationProvider>
                  <FourthStationProvider>
                      <FifthStationProvider>
                          <SixthStationProvider>
                              <Stations />
                          </SixthStationProvider>
                      </FifthStationProvider>
                  </FourthStationProvider>
              </ThirdStationProvider>
          </SecondStationProvider>
      </FirstStationProvider>, div
    )
  })

  it('should have 6 images with classname station', () => {
    expect(wrapperMount.find('img.station').length).toEqual(6)
  })

  test('so FirstStationContext will set position', () => {
    const position = {bottom: '380px', left: '120px'}
    renderStations(position)
    expect(screen.getByAltText('Science station for North America').style.bottom).toEqual('380px')
    expect(screen.getByAltText('Science station for North America').style.left).toEqual('120px')
  })

  test('so SecondStationContext will set position', () => {
    const position = {bottom: '380px', left: '120px'}
    renderStations(null, null, position)
    expect(screen.getByAltText('Science station for South America').style.bottom).toEqual('380px')
    expect(screen.getByAltText('Science station for South America').style.left).toEqual('120px')
  })

  test('so ThirdStationContext will set position', () => {
    const position = {bottom: '380px', left: '120px'}
    renderStations(null, null, null, null, position)
    expect(screen.getByAltText('Science station for Africa').style.bottom).toEqual('380px')
    expect(screen.getByAltText('Science station for Africa').style.left).toEqual('120px')
  })

  test('so FourthStationContext will set position', () => {
    const position = {bottom: '380px', left: '120px'}
    renderStations(null, null, null, null, null, null, position)
    expect(screen.getByAltText('Science station for Europe').style.bottom).toEqual('380px')
    expect(screen.getByAltText('Science station for Europe').style.left).toEqual('120px')
  })

  test('so FifthStationContext will set position', () => {
    const position = {bottom: '380px', left: '120px'}
    renderStations(null, null, null, null, null, null, null, null, position)
    expect(screen.getByAltText('Science station for Asia').style.bottom).toEqual('380px')
    expect(screen.getByAltText('Science station for Asia').style.left).toEqual('120px')
  })

  test('so SixthStationContext will set position', () => {
    const position = {bottom: '380px', left: '120px'}
    renderStations(null, null, null, null, null, null, null, null, null, null, position)
    expect(screen.getByAltText('Science station for Australia').style.bottom).toEqual('380px')
    expect(screen.getByAltText('Science station for Australia').style.left).toEqual('120px')
  })
})