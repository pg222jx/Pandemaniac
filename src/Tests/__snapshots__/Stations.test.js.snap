// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Stations component should match the snapshot 1`] = `"<img src=\\"station.png\\" alt=\\"Science station for North America\\" id=\\"firstStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for South America\\" id=\\"secondStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Africa\\" id=\\"thirdStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Europe\\" id=\\"fourthStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Asia\\" id=\\"firthStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/><img src=\\"station.png\\" alt=\\"Science station for Australia\\" id=\\"sixthStation\\" class=\\"station\\" style=\\"visibility:hidden\\"/>"`;
